### PR TITLE
Refactor DSC config to WEX shared config

### PR DESCRIPTION
**Pull request by Cruikshanks**
A recent issue with the service exposed that some of our config had gotten 'in a muddle' following the merging of the core gems into this project. This was not helped by us having a separate Digital services core config and a WEX config, both of which used different mechanisms and ways of initialising the values.

This change covers the refactoring of all logic related to the `DigitalServicesCore::Configuration` into `WasteExemptionsShared::Configuration`.

- Moved config properties into `lib/waste_exemptions_shared/configuration.rb`
- Moved initialisation of these properties into lib/waste_exemptions_shared/engine.rb
- Deleted the `digital_services_core_config` initializer from the dummy app
- Updated all references to `WasteExemptionsShared::Configuration` to use a consistent syntax
- Updated spec helpers to use the new config object rather then the old DSC one

Along with these changes there was also some housekeeping

- Updated .env.example to cover missing environment variables
- Corrected dummy apps `secrets.yml`

There is still some work to go. We have `WasteExemptionsShared::Configuration` for handling config, but don't always use it (see how we grab email setup values in the environment files), but this is beyond the scope of this initial change.

### CODE REVIEW

**Comment by Cruikshanks** [comment]
This also covers fixing the admin area lookup issue hence I have closed #131 .

**Comment by Cruikshanks** [comment]
Scratch that, the admin lookup is not actually working. Spotted it's because I still need to move `app/services/digital_services_core/resolve_adminsitrative_area.rb` to `app/services/waste_exemptions_shared`. However as this has been missed again I think I really need to get a test in before continuing.

File: app/views/digital_services_core/shared/_google_tag_manager.html.erb [hunk 1]
```diff
@@ -7,5 +7,5 @@
   new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
   j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
   '//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
-  })(window,document,'script','dataLayer','<%= WasteExemptionsShared.google_tag_manager_id %>');</script>
+})(window,document,'script','dataLayer','<%= WasteExemptionsShared.config.google_tag_manager_id %>');</script>
```
Comment (TThurston) [inline app/views/digital_services_core/shared/_google_tag_manager.html.erb]: Was change in indentation intentional?

File: lib/waste_exemptions_shared/engine.rb [hunk 2]
```diff
@@ -56,15 +56,31 @@ class Engine < ::Rails::Engine
...
+        # Attempt to pick up the Google Tag Manager ID from an environment variable.
+        # If the variable is not set, we cannot use Google Analytics.
+        config.google_tag_manager_id = ENV["DSC_FRONTEND_GOOGLE_TAGMANAGER_ID"]
+
+        # Tracking using Google Analytics. As noted above, we can only do this if we
+        # know the Google Tag Manager ID.  Additionally, whilst we want to do this
+        # in production, it is optional elsewhere.
+        unless config.google_tag_manager_id.blank?
+          config.use_google_analytics =
+            (ENV["DSC_FRONTEND_USE_GOOGLE_ANALYTICS"] == "true") || Rails.env.production? || false
```
Comment (TThurston) [inline lib/waste_exemptions_shared/engine.rb]: `|| false` isn't necessary, I think, but may be nice to keep for clarity?

File: lib/waste_exemptions_shared/engine.rb [hunk 3]
```diff
@@ -56,15 +56,31 @@ class Engine < ::Rails::Engine
...
+        end
+
+        config.services_phone =
+          ENV["DSC_SERVICES_PHONE"] || fail("Environment variable DSC_SERVICES_PHONE has not been set")
+
         config.backto_start_url = "https://www.gov.uk/guidance/register-your-waste-exemptions-environmental-permits"
```
Comment (TThurston) [inline lib/waste_exemptions_shared/engine.rb]: Any value in moving this to an environment variable, or allowing this default to be overridden in that manner?

File: spec/dummy/config/secrets.yml [hunk 4]
```diff
@@ -12,7 +12,7 @@ defaults: &defaults
 
 
 
-defaults: &defaults
+#defaults: &defaults
```
Comment (TThurston) [inline spec/dummy/config/secrets.yml]: Was this change intentional?

File: lib/waste_exemptions_shared/engine.rb [hunk 5]
```diff
@@ -56,15 +56,31 @@ class Engine < ::Rails::Engine
 
     initializer :wex_shared_config do
       WasteExemptionsShared.configure do |config|
-        config.exemptions_expire_after_duration = 3.years - 1.day
```
Comment (TThurston) [inline lib/waste_exemptions_shared/engine.rb]: 👍

**Comment by TThurston** [comment]
Suggesting @lewispb also casts his eyes over this for a second opinion.

File: app/views/digital_services_core/shared/_google_tag_manager.html.erb (hunk 1 above)
Comment (Cruikshanks) [inline app/views/digital_services_core/shared/_google_tag_manager.html.erb]: Nope, didn't spot that had happened. Will fix.

File: lib/waste_exemptions_shared/engine.rb (hunk 2 above)
Comment (Cruikshanks) [inline lib/waste_exemptions_shared/engine.rb]: Think that was me just panicking. Also don't think its necessary so shouldn't be there. Will fix.

File: lib/waste_exemptions_shared/engine.rb (hunk 3 above)
Comment (Cruikshanks) [inline lib/waste_exemptions_shared/engine.rb]: On the basis that anything that is likely to change across environments and/or deployments should be an environment variable ([12 factor app](https://12factor.net/config) then yes I agree. However I would put it outside the scope of this change, which was to merge previous DSC config into the WasteExemptionsShared namespace, and fix the admin lookup (though you could argue even that should have been done as a separate PR).

If it was just switching it to an env var that would be great, but it also means updating all for environments to include the new env var, and that can be a mini project in itself so I would rather handle it separately.

File: spec/dummy/config/secrets.yml (hunk 4 above)
Comment (Cruikshanks) [inline spec/dummy/config/secrets.yml]: Yes, but it should be deleted, not just commented out. Good spot, will fix. 😳

**Comment by Cruikshanks** [comment]
I'll push the corresponding front end changes so you have some context @TThurston

**Comment by Cruikshanks** [comment]
The corresponding front end PR's are pushed. You're welcome to start looking through them now, but until this is merged they can't be said to be complete as I need to update the references in their `Gemfile`s.

- <https://github.com/EnvironmentAgency/waste-exemptions/pull/251>
- <https://github.com/EnvironmentAgency/waste-exemptions-back-office/pull/152>
